<a href="https://colab.research.google.com/github/bruh7463/data_mining_project_team_14/blob/main/data_mining_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<center>
  <font size=6><b>CSC 4792: Data Mining and Warehousing</b></font><br />
  <font size=3>Group Project: Classify country of origin for Wikipedia user contributions on Zambian Wikipedia pages</font>

   <b>Group Members</b> </br>
   Simon Chanda </br>
   Lawrence Chitandula </br>
   Chilufya Chulu </br>
   Kasweka Michael Mukoko </br>
   Mike Shatamuka </br>
</center>


# **1. Business Understanding**
### Problem Statement
Wikipedia pages related to Zambia have fewer local contributions than international ones. There is little understanding of where the contributors of Zambian-related pages come from, including the Zambia page. Without this information, it is difficult to create specific strategies that boost contributions from Zambian Wikipedians. The goal is to find out and categorize the origins of Wikipedia users who contribute to Zambian pages. This will help in planning actions that promote more local involvement.

### Business Objectives
- Identify the percentage of contributions to Zambian Wikipedia articles by users from Zambia compared to other nations.
- Provide practical advice to guide strategies for Zambian-based contributions.
- Support The University of Zambia's DataLab Research Group in advocating for greater representation of local perspectives on Wikipedia.
- Enable long-term monitoring of contribution by country of origin.

### What Success Looks Like In Practice
- An unambiguous evidence based report on the contributions by country.
- It was possible to identify trends of where most of edits are made.
- Lessons that could be shared with Wikimedia communities and policymakers incase of encouraging locals to contribute.

### Data Mining Goals
- Create a prediction model that can determine the most likely country of origin for each Wikipedia edit made to Zambian pages, using details like the user’s IP address (for anonymous edits), profile information, and the timing of edits.

- Analyze and visualize the data to show clear summaries of how contributions are distributed by country.

- Spot unusual trends or activity, such as sudden increases in edits coming from a specific country.

### Initial Project Success Criteria

- Produce a clear and accurate breakdown of Wikipedia contributions to Zambian pages by country of origin.
- Achieve at least 80% accuracy in classifying the country of origin for contributions.
- Visualize contribution patterns and trends over time through interpretable dashboards or charts.
- Provide actionable insights that can guide strategies to increase local Zambian participation.



# **2. Data Understanding**

In [13]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [14]:
import pandas as pd

# Load CSV
file_path = "/content/drive/MyDrive/Dataset/zambia_wikipedia_history.csv"
df = pd.read_csv(file_path)

# Preview dataset
print("Dataset Shape:", df.shape)
df.head()

Dataset Shape: (528, 10)


page_id   title  rev_id             timestamp  contributor_id  \
0    15643  Zambia   66041  2005-10-31T08:48:31Z             NaN   
1    15643  Zambia   66727  2005-11-05T06:07:50Z             NaN   
2    15643  Zambia  100931  2006-02-14T12:33:52Z             NaN   
3    15643  Zambia  100964  2006-02-14T15:58:40Z             NaN   
4    15643  Zambia  100987  2006-02-14T16:17:15Z             NaN   

                       contributor_name  contributor_ip               comment  \
0        User(id=1734, text='Ianleow7')             NaN     Creating article.   
1      User(id=1110, text='Ricky81682')             NaN         rm unused cat   
2  User(id=None, text='81.204.151.102')             NaN                   NaN   
3         User(id=2131, text='Archer7')             NaN  corrected formatting   
4      User(id=2077, text='Freshstart')             NaN     typos, full names   

   minor  text_length  
0  False         1617  
1  False         1577  
2  False         2211  
3  False         2219  
4   True         2273

- The dataset contains 528 rows and 10 columns.

- First look shows columns like: title, rev_id, timestamp, contributor, text_length.

- Data seems to have both text and numeric attributes.